index.php

```diff
@@ -78,9 +83,16 @@ function handle($data) {
     $tmp = trim(strtolower($data['username']));
     $cloudUsername = substr($tmp, 0, strpos($tmp, '@'));
     $cloudDomain = substr($tmp, strpos($tmp, '@') + 1);
+    // if there is @qrcode in the string, password is a token. Remove it and set the isToken to true.
+    if (strpos($data['username'], '@qrcode') !== false) {
```

Comment:
Maybe it's better to check if the well-known string is at the end
```suggestion
    if (str_ends_with($data['username'], '@qrcode') ) {
```

Reply:
I used strpos because on testing apps, it should be possible that the string ends with @NETHVOICEAPP. In that case, the format would be USERNAME@DOMAIN:<PASSWORD|TOKEN@qrcode>@APPNAME

Reply:
ok, still not best solution but it should the work ... unless a user is named qrcode :D